components/brave_wallet/browser/fil_transaction.cc

```diff
@@ -104,22 +111,23 @@ absl::optional<FilTransaction> FilTransaction::FromTxData(
 
 base::Value FilTransaction::ToValue() const {
   base::Value dict(base::Value::Type::DICTIONARY);
-  dict.SetStringKey("nonce",
-                    nonce_ ? base::NumberToString(nonce_.value()) : "");
-  dict.SetStringKey("gas_premium", gas_premium_);
-  dict.SetStringKey("gas_fee_cap", gas_fee_cap_);
-  dict.SetStringKey("max_fee", max_fee_);
-  dict.SetStringKey("gas_limit", base::NumberToString(gas_limit_));
-  dict.SetStringKey("to", to_.EncodeAsString());
-  dict.SetStringKey("value", value_);
+  dict.SetStringKey("Nonce",
```

Comment:
If I'm understanding how the nonce value get's to this point I think this is going to potentially lead to problems. Ideally we should be using a CSPRNG here to generate nonces I believe and we shouldn't default to a `0` here and rather should error if one can't be generated. Since BLS signatures are deterministic signatures this could accidentally lead to a replay attack with them.

When I was looking through this code a bit further to understand where the Nonce value is coming from it seemed like it was coming all the way from UI layer (in the typescript code) as well and optionally passed down here. Is there a reason we'd need the UI layer to be able to provide that nonce value or would it be ok if we just did the nonce generation at this code point and then remove the optionality from the UI down to here.

Reply:
The Nonce value is coming from server side as result of `Filecoin.MpoolGetNonce` request. It is increased for each transaction per address. It can be visible in some UI locations as read only part of information about a message.

Reply:
Hmm ok, I didn't catch that so thanks for redirecting me in the right direction. I'm going to spend a bit more time looking into this then just to think through if that nonce being server side or being tampered with could potentially introduce some other issues.

Reply:
👍🏻 ok I've spent some time digging through the docs for [Lotus' JSON-RPC API](https://lotus.filecoin.io/developers/apis/json-rpc/#mpoolgetnonce) and the [postman collection](https://documenter.getpostman.com/view/4872192/SWLh5mUd#0656cf16-a0e0-4fbc-bf62-c6bc3011a10f) from the looks of it this is fine. My understanding is the nonce is maintained per address and this call is made ahead of time to get the current nonce of the address from the node, so I don't forsee any issues here.